packages/deploy/src/stateTransform.ts

```diff
@@ -130,6 +149,16 @@ function mergeTriggers(
               ...(specTrigger.type === 'cron'
                 ? { cron_expression: specTrigger.cron_expression }
                 : {}),
+              ...(specTrigger.type === 'kafka'
```

Comment:
This is getting way to complicated and I'm going to insist on a refactor here :(

I've just scribbled this and haven't tested it, but something along these lines would be a lot more readable and scalable:
```
 if (specTrigger && !stateTrigger) {
    const trigger = {
      id: crypto.randomUUID(),
      ...pickKeys(specTrigger, ['type', 'enabled']),
    };
    
    if (specTrigger.type === 'cron') {
      trigger.cron_expression = specTrigger.cron_expression
    } else if (specTrigger.type === 'kafka') {
      trigger.kafka_configuration = {
        ...specTrigger.kafka_configuration,
        hosts: transformSpecKafkaHost(
          specTrigger.kafka_configuration?.hosts ?? []
        ),
      },
    }

    return [triggerKey, trigger]
  }
```